src/dotnet/CarbonAware.WebApi.Tests/unitTests/SciScoreControllerTests.cs

```diff
@@ -0,0 +1,136 @@
+    /// </summary>
+    [TestCase(LocationType.Geoposition)]
+    [TestCase(LocationType.CloudProvider)]
+    public async Task SuccessfulCallReturnsOk_MarginalCarbonIntensity(LocationType locationType)
+    {
+        double data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+        LocationInput location = new LocationInput()
+            { 
+                LocationType = locationType.ToString()
+            };
+        string timeInterval = "2007-03-01T13:00:00Z/2007-03-01T15:30:00Z";
+        SciScoreInput input = new SciScoreInput()
+        {
+            Location = location,
+            TimeInterval = timeInterval
+        };
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
+        var expected = new SciScore() { MarginalCarbonEmissionsValue = data };
+
+        TestHelpers.AssertStatusCode(carbonIntensityOutput, HttpStatusCode.OK);
+        Assert.AreEqual(expected, carbonIntensityOutput.Value);
+    }
+
+    /// <summary>
+    /// Tests that invalid time inputs respond with a badRequest error
+    /// </summary> location 
+    [Test]
+    public async Task InvalidTimeIntervalReturnsBadRequest_MarginalCarbonIntensity()
+    {
+        // Arrange
+        var data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+
+        LocationInput locationInput = new LocationInput()
+        {
+            LocationType = LocationType.Geoposition.ToString()
+        };
+
+        string timeInterval = "";
+
+        SciScoreInput input = new SciScoreInput()
+        {
+            Location = locationInput,
+            TimeInterval = timeInterval
+        };
+
+        // Act
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
+
+        // Assert
+        var expected = new CarbonAwareWebApiError() { Message = "timeInterval is required" };
+        TestHelpers.AssertStatusCode(carbonIntensityOutput, HttpStatusCode.BadRequest);
+        Assert.AreEqual(expected, carbonIntensityOutput.Value);
+    }
+
+    /// <summary>
+    /// Tests that invalid location inputs respond with a badRequest error
+    /// </summary> location 
+    [Test]
+    public async Task NullLocationReturnsBadRequest_MarginalCarbonIntensity()
+    {
+        // Arrange
+        var data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+
+        string timeInterval = "2007-03-01T13:00:00Z/2007-03-01T15:30:00Z";
+
+        SciScoreInput input = new SciScoreInput()
+        {
+            TimeInterval = timeInterval
+        };
+
+        // Act
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
```

Comment:
I would consider adding a set of checks around this to avoid the error the static analysis tool is calling out below